### DISCUSSION TITLE
Quickstart - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#2362] [Docs]
# Quickstart - Ultralytics YOLOv8 Docs

http://127.0.0.1:8000/quickstart/

**Comment by pderrenger** [comment]
Would it be possible to add a social media "share" button to the docs?

**Reply by glenn-jocher** [reply]
@pderrenger yes, all done 😃

**Comment by sevid42** [comment]
Hi there, nice tutorial, could you upload something for EC2 to train and get inferences? 
Thanks

**Reply by glenn-jocher** [reply]
@sevid42 yes we have an AWS EC2 quickstart tutorial at https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/

**Comment by jakerankincgi** [comment]
Hi,

I get an error:

AttributeError: module 'datasets' has no attribute 'Split' 

How can I get around this? This occurs at the model.predict part and leads to here:

class HuggingFaceDatasetSource(DatasetSource):
     11     """
     12     Represents the source of a Hugging Face dataset used in MLflow Tracking.
     13     """
     15     def __init__(
     16         self,
     17         path: str,
     18         config_name: Optional[str] = None,
     19         data_dir: Optional[str] = None,
     20         data_files: Optional[
     21             Union[str, Sequence[str], Mapping[str, Union[str, Sequence[str]]]]
     22         ] = None,
---> 23         split: Optional[Union[str, datasets.Split]] = None,

**Reply by glenn-jocher** [reply]
The code you have above is not part of the Ultralytics package, so you should raise a bug report with the authors of the code creating the error.

**Comment by zjy1220** [comment]
Hi ，
I get an error too!
When I executed the following training command according to the tutorial, an error occurred
yolo train data=coco128.yaml model=yolov8n.pt epochs=10 lr0=0.01

Ultralytics YOLOv8.0.115 🚀 Python-3.8.16 torch-2.0.1+cu117 CUDA:0 (NVIDIA GeForce RTX 2070, 7941MiB)
yolo/engine/trainer: task=detect, mode=train, model=yolov8n.pt, data=coco128.yaml, epochs=10, patience=50, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=False, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=0, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, v5loader=False, tracker=botsort.yaml, save_dir=runs/detect/train
Downloading https://ultralytics.com/assets/Arial.ttf to /home/zjy/.config/Ultralytics/Arial.ttf...
100%|██████████████████████████████████████████████| 755k/755k [00:01<00:00, 762kB/s]

                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 
 22        [15, 18, 21]  1    897664  ultralytics.nn.modules.head.Detect           [80, [64, 128, 256]]          
Model summary: 225 layers, 3157200 parameters, 3157184 gradients

Transferred 355/355 items from pretrained weights
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
train: Scanning /home/zjy/work/ultralytics/datasets/coco128/labels/train2017.cache...
val: Scanning /home/zjy/work/ultralytics/datasets/coco128/labels/train2017.cache... 1
Plotting labels to runs/detect/train/labels.jpg... 
optimizer: AdamW(lr=0.000119, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to runs/detect/train
Starting training for 10 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       1/10      2.51G      1.226      1.615      1.274        178        640: 100%|█
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-
Traceback (most recent call last):
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/bin/yolo", line 8, in <module>
    sys.exit(entrypoint())
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/cfg/__init__.py", line 398, in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/engine/model.py", line 371, in train
    self.trainer.train()
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/engine/trainer.py", line 192, in train
    self._do_train(world_size)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/engine/trainer.py", line 370, in _do_train
    self.metrics, self.fitness = self.validate()
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/engine/trainer.py", line 476, in validate
    metrics = self.validator(self)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/engine/validator.py", line 160, in __call__
    preds = model(batch['img'], augment=self.args.augment)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/nn/tasks.py", line 45, in forward
    return self.predict(x, *args, **kwargs)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/nn/tasks.py", line 62, in predict
    return self._predict_once(x, profile, visualize)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/nn/tasks.py", line 82, in _predict_once
    x = m(x)  # run
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/nn/modules/block.py", line 180, in forward
    y = list(self.cv1(x).chunk(2, 1))
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/nn/modules/conv.py", line 38, in forward
    return self.act(self.bn(self.conv(x)))
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/nn/modules/conv.py", line 463, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/nn/modules/conv.py", line 459, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: GET was unable to find an engine to execute this computation


Thank you all very much for your assistance.

**Reply by zjy1220** [reply]
Some users suggested that installing these three versions can solve the problem: torch 1.13.0, torchvision 0.14.0, and torchaudio 0.13.0. However, after installing the specified versions, I encountered new errors in both predict and training, indicating potential library compatibility issues. The specific error log information is as follows:

yolo train data=coco128.yaml model=yolov8n.pt epochs=10 lr0=0.01
Traceback (most recent call last):
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/bin/yolo", line 5, in <module>
    from ultralytics.yolo.cfg import entrypoint
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/__init__.py", line 5, in <module>
    from ultralytics.hub import start
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/hub/__init__.py", line 5, in <module>
    from ultralytics.hub.auth import Auth
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/hub/auth.py", line 5, in <module>
    from ultralytics.hub.utils import HUB_API_ROOT, PREFIX, request_with_credentials
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/hub/utils.py", line 14, in <module>
    from ultralytics.yolo.utils import (ENVIRONMENT, LOGGER, ONLINE, RANK, SETTINGS, TESTS_RUNNING, TQDM_BAR_FORMAT,
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/__init__.py", line 3, in <module>
    from . import v8
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/v8/__init__.py", line 3, in <module>
    from ultralytics.yolo.v8 import classify, detect, pose, segment
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/v8/classify/__init__.py", line 3, in <module>
    from ultralytics.yolo.v8.classify.predict import ClassificationPredictor, predict
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/ultralytics/yolo/v8/classify/predict.py", line 3, in <module>
    import torch
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/__init__.py", line 191, in <module>
    _load_global_deps()
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/__init__.py", line 153, in _load_global_deps
    ctypes.CDLL(lib_path, mode=ctypes.RTLD_GLOBAL)
  File "/home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/ctypes/__init__.py", line 373, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/zjy/Downloads/ENTER/envs/yolov8_env/lib/python3.8/site-packages/torch/lib/../../nvidia/cublas/lib/libcublas.so.11: symbol cublasLtGetStatusString version libcublasLt.so.11 not defined in file libcublasLt.so.11 with link time reference

**Reply by glenn-jocher** [reply]
@zjy1220 hello,

It seems you're encountering compatibility issues with the specific versions of PyTorch, torchvision, and torchaudio. The error message indicates a problem with the CUDA libraries, particularly with `libcublas`. This could be due to a mismatch between the CUDA version installed on your system and the one expected by the PyTorch binaries you've installed.

Here are a few steps you can take to troubleshoot the issue:

1. Ensure that your CUDA toolkit version matches the version required by the PyTorch installation. You can check the compatibility on the official PyTorch website.

2. If you have multiple CUDA versions installed, make sure that the `PATH` and `LD_LIBRARY_PATH` environment variables are set to point to the correct version of CUDA.

3. Consider creating a fresh virtual environment and reinstalling the correct versions of PyTorch, torchvision, and torchaudio that are compatible with your CUDA version.

4. If the issue persists, you might want to try installing PyTorch using the pre-built binaries from the official PyTorch website, which are built against specific CUDA versions.

5. Lastly, check the Ultralytics documentation for any additional guidance on setting up the environment for YOLOv8.

Remember to always match the CUDA toolkit version with the PyTorch binaries to avoid such errors. If you continue to face issues, please provide more details about your environment, including the CUDA and cuDNN versions, and we'll do our best to assist you further.

**Comment by HermanCoding** [comment]
Hi 
Are you somehow able to download the coco dataset, using ultralytics, without starting to train a model?

**Reply by glenn-jocher** [reply]
No, but it's very simple to run a YOLO COCO training command that will automatically download the dataset:
```bash
yolo train data=coco.yaml
```

**Comment by qyq-20010530** [comment]
Hello, 
I get an error:
RuntimeError: Given groups=1, weight of size [96, 96, 1, 1], expected input[1, 48, 128, 160] to have 96 channels, but got 48 channels instead.
This is an error caused by replacing the weight file yolov5s.pt with yolov5m.pt. I don't know how to solve it, and I sincerely hope you can give me some guidance.

**Reply by hannaliavoshka** [reply]
Hi, it's hard to say what is wrong from the general description, can you please specify how have you replaced the weights file? 
Using 'model = YOLO(yolov5m.pt)' or another way? Is it YOLO default weights file or pre-trained on your data?

**Reply by glenn-jocher** [reply]
@hannaliavoshka hello! It seems like you're encountering a channel mismatch error, which typically occurs when the model architecture expects a different number of input channels than what is being provided. This can happen if you're trying to load weights from a model that has a different configuration or number of layers.

If you're attempting to switch from a smaller model like YOLOv5s to a larger one like YOLOv5m, ensure that you're using the correct model definition that matches the weights file. Each model variant (small, medium, large, etc.) has a specific architecture, so their weights are not interchangeable without modifying the model definition to match.

Make sure you're initializing the correct model architecture before loading the weights. If you're using pre-trained weights, they must correspond to the model's architecture you're trying to use.

For more detailed guidance, please refer to our documentation on model training and usage at https://docs.ultralytics.com. If you continue to face issues, feel free to provide more context or check the existing issues in the repo for similar problems that have been resolved. Good luck! 🚀

**Comment by akshay9396** [comment]
Hello, 
I am trying to run Dockerfile_arm64 on Raspberry pi. I am getting error  with temporary failure. Is there any Readme  available.
Could you please share ?

**Reply by glenn-jocher** [reply]
@akshay9396 hello! I'm glad to hear you're working with YOLOv8 on a Raspberry Pi. For running the Dockerfile on an ARM64 architecture like the Raspberry Pi, please ensure that you have the latest version of Docker installed and that your system meets the necessary prerequisites for running YOLOv8.

While we don't have a specific Readme for the ARM64 Dockerfile, the general instructions for setting up and running YOLOv8 can be found in our documentation. Please visit the Ultralytics Docs and review the installation and setup guides, which should provide you with the necessary steps to get started.

If you continue to experience issues, please provide more details about the error message you're receiving, and we'll do our best to assist you further. Remember to check that your Raspberry Pi has sufficient resources (memory, CPU) to handle the demands of YOLOv8, as this can sometimes cause temporary failures during setup or operation. 🚀

**Comment by akshay9396** [comment]
Hello,
I am using the default weight file.
Is there any instruction module available to run docker image of yolov8 on
Raspberry Pi?

I'm looking forward to hearing from you soon.

Best regards,
Akshay Kanade

**Reply by glenn-jocher** [reply]
@akshay9396 hello Akshay,

Thank you for reaching out. Currently, we do not have a dedicated instruction module for running a YOLOv8 Docker image on a Raspberry Pi. However, you can still attempt to run YOLOv8 on a Raspberry Pi by ensuring you have Docker installed and then pulling the appropriate Docker image from our repository.

Keep in mind that running YOLOv8 on a Raspberry Pi may present performance limitations due to the hardware constraints of the device. For optimal performance, we recommend using a more powerful system.

If you have further questions or need assistance with specific steps, please refer to our documentation at https://docs.ultralytics.com, which provides comprehensive guidance on various modes and tasks that YOLOv8 supports.

**Comment by qyq-20010530** [comment]
I am very pleased to receive your reply. I have resolved this issue. I downloaded YOLOv5 version 6.1 again and downloaded the corresponding weight files, placing them in the vehicle detection and counting project. After running the program, there are no more errors! Thank you very much for your reply！





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ultralytics/ultralytics"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年7月18日(星期二) 凌晨1:42
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ultralytics/ultralytics] Quickstart - Ultralytics YOLOv8 Docs (Discussion #2362)





 
Hi, it's hard to say what is wrong from the general description, can you please specify how have you replaced the weights file?
 Using 'model = YOLO(yolov5m.pt)' or another way? Is it YOLO default weights file or pre-trained on your data?

**Reply by glenn-jocher** [reply]
@qyq-20010530 i'm glad to hear that you've resolved the issue with YOLOv5. Just a gentle reminder that this repository is focused on YOLOv8, which is the latest iteration of the YOLO series. If you have any questions or need assistance with YOLOv8 in the future, feel free to reach out. The YOLO community and the Ultralytics team are always here to help. Happy coding! 😊🚀

**Comment by akshay9396** [comment]
Hello,
just the last question, is the YOLOv5 docker image  working on Raspberry Pi?
If yes, Could you please send the link of docker image working on Raspberry
pi.

Looking forward to hearing from you soon.

Best regards,
Akshay kanade

**Reply by glenn-jocher** [reply]
@akshay9396 hello Akshay,

Thank you for reaching out! Regarding the YOLOv5 Docker image, I'd like to gently remind you that this repository is focused on YOLOv8. For YOLOv5-specific inquiries, I recommend checking the YOLOv5 repository and its documentation.

As for running on a Raspberry Pi, Docker images need to be built specifically for ARM architectures, which the Raspberry Pi uses. If you're looking to run YOLO models on a Raspberry Pi, you should look for a Docker image that is compatible with ARMv7 (for 32-bit OS) or ARMv8 (for 64-bit OS).

For more information on YOLOv8 and its capabilities, please refer to our documentation at https://docs.ultralytics.com. If you have further questions about YOLOv8, feel free to ask!

**Comment by phodgers** [comment]
For the complete newbie, I think you can improve this quickstart (and git repo instructions) a little.
1) For the CLI section, I think you should have it defaulting to the predict tab, so that the example is right there to run (the same as you give in the github repo).
2) When the user runs "yolo predict model=yolov8n-seg.pt source='https://youtu.be/Zgi9g1ksQHc' imgsz=320", the output says "Results saved to runs\detect\predict". But you could also add a line to say enter "%cd%\runs\detect\predict\bus.jpg" (windows) to view the result and similar one liners for Linux/Mac.
3) I found my older version of Python 3.9 had issues running (even though install said dependencies were met, it then complained about numpy due to version issues), after upgrading to current Python 3.11.5, installation all went smoothly, so might be worth adding a suggestion at the beginning of the quickstart to also update to latest version of Python before the "pip install ultralytics".

**Reply by glenn-jocher** [reply]
@phodgers thank you for your valuable feedback! We always appreciate suggestions that can help improve the user experience.

1) We'll consider making the predict tab the default view in the CLI section of the Quickstart guide to provide immediate access to a practical example.

2) Noted! We'll look into adding a concise line in the output to guide users on how to view their results directly from the command line for different operating systems.

3) Regarding Python compatibility, we'll review our documentation to ensure we communicate the recommended Python version more clearly to avoid installation issues related to version compatibility.

Your input is instrumental in making our documentation more user-friendly. If you have any more suggestions or need further assistance, please feel free to reach out. Happy coding! 😊🚀
